### PR TITLE
drm: Replace inner Option on DrmNode and use mem::forget

### PR DESCRIPTION
**Pull request by i509VCB**
Let's clean up the internals of DrmNode a bit.

### CODE REVIEW

File: src/backend/drm/node/mod.rs [hunk 1]
```diff
@@ -120,22 +115,22 @@ impl Display for DrmNode {
 }
 
 impl IntoRawFd for DrmNode {
-    fn into_raw_fd(mut self) -> RawFd {
-        self.fd.take().unwrap()
+    fn into_raw_fd(self) -> RawFd {
+        let fd = self.fd;
+        mem::forget(self);
```
Comment (Drakulix) [inline src/backend/drm/node/mod.rs:120]: But doesn't that leak memory?

Comment (i509VCB) [reply src/backend/drm/node/mod.rs:120]: Nope:
> Takes ownership and “forgets” about the value without running its destructor.
>
> Any resources the value manages, such as heap memory or a file handle, will linger forever in an unreachable state. > However, it does not guarantee that pointers to this memory will remain valid.

Comment (i509VCB) [reply src/backend/drm/node/mod.rs:120]: Plus the semantics defined by IntoRawFd indicate the caller takes ownership of the open fd

Comment (Drakulix) [reply src/backend/drm/node/mod.rs:120]: Yes for the Fd, that is correct, but we leak the memory of "self".

Comment (Drakulix) [reply src/backend/drm/node/mod.rs:120]: Misunderstood, stack variables are of course still deallocated, just not "dropped"

**Review by Drakulix — APPROVED** [review]

**Pull request merged**